File: data-structure-algorithms/sorting/selection-sort.go
package main

import (
	"fmt"
	"math/rand"
	"time"
)

func SelectionSort(a []int) {

	for i := 0; i < len(a); i++ {

		currentIndex := i

		for j := i + 1; j < len(a); j++ {
			if a[currentIndex] > a[j] {
				temp := a[currentIndex]
				a[currentIndex] = a[j]
				a[j] = temp
			}
		}
	}

}

func main() {

	arr := make([]int, 5)
	source := rand.NewSource(time.Now().UnixNano())
	r := rand.New(source)

	for idx, _ := range arr {
		// Generate a random integer between 0 and 99 (exclusive of 100)
		random := r.Intn(100)
		arr[idx] = random
	}

	SelectionSort(arr)

	for _, it := range arr {
		fmt.Println(it, " ")
	}

}

File: data-structure-algorithms/sorting/selection-sort.cpp
#include<iostream>
using namespace std;

int main(){
    const int n = 9;
    int a[n] = {4, 5, 35, 2, 553, 5, 2, 5, 3};

    for (int i = 0; i < n; i++){
        cout << " "<< a[i];
    }

    for (int i = 0; i < n-1;i++){
        int currentIndex = i ;
        for (int j = i + 1; j < n;j++){
            if(a[currentIndex]> a[j]){
                swap(a[currentIndex], a[j]);
            }
        }
    }

    cout << endl;
    for (int i = 0; i < n; i++)
    {
        cout << " "<< a[i];
    }

        return 0;
}